### PR TITLE
fix state transition on invoice print

### PR DESCRIPTION
**Pull request by patrickweh**
add MenuManager and facade

### CODE REVIEW

File: src/Facades/Menu.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @see \FluxErp\Menu\MenuManager
+ */
+class Menu extends Facade
+{
+    protected static function getFacadeAccessor(): string
+    {
+        return 'flux.menu_manager';
+    }
+
+    public static function register(
```
Comment (SirSplasch) [inline src/Facades/Menu.php:18]: this function can be removed, or am I wrong?

File: src/Menu/MenuManager.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            function (array $value) use ($guard, $ignorePermissions) {
+                if (($value['guard'] ?? $guard) !== $guard) {
+                    return false;
+                }
+
+                // first check if a permission exists
+                if (($value['permission'] ?? false) && ! $ignorePermissions) {
+                    // check if a permission for the current guard exists
+                    if ($value['permission'] === null) {
+                        dd($value);
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php:98]: why is there a dd()?

File: src/Menu/MenuManager.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+namespace FluxErp\Menu;
+
+use FluxErp\Models\Permission;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Illuminate\Support\Traits\Macroable;
+use Spatie\Permission\Exceptions\PermissionDoesNotExist;
+
+/**
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php:12]: DocBlock can be removed, this is not the facade, this is the concrete implementation

File: src/Menu/MenuManager.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            [
+                'label' => $label ?: Str::afterLast($path, '.'),
+                'uri' => Str::of($route->uri())->start('/')->toString(),
+                'icon' => $icon,
+                'route_name' => $routeName,
+                'guard' => $guard,
+                'domain' => $route->getDomain(),
+                'permission' => $route->getPermissionName(),
+                'order' => $order
+                    ?? count(
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php:67]: ?:

File: src/Menu/MenuManager.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                        data_get($this->menuItems, Str::beforeLast($path, '.'))
+                            ?? (str_contains($path, '.') ? [] : $this->menuItems
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php]: ?:

File: src/Menu/MenuManager.php [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                }
+
+                // first check if a permission exists
+                if (($value['permission'] ?? false) && ! $ignorePermissions) {
+                    // check if a permission for the current guard exists
+                    if ($value['permission'] === null) {
+                        dd($value);
+                    }
+                    try {
+                        $permission = Permission::findByName($value['permission'], $guard);
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php:101]: unused variable

File: src/Menu/MenuManager.php [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+                    } catch (PermissionDoesNotExist $e) {
```
Comment (SirSplasch) [inline src/Menu/MenuManager.php:102]: unused variable

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/Menu/MenuManager.php (hunk 4 above)
Comment (patrickweh) [reply src/Menu/MenuManager.php:67]: If Position is 0 ?: triggers the "Else", ?? Only if null

File: resources/views/livewire/navigation.blade.php [hunk 8]
```diff
@@ -85,22 +85,22 @@ class="dark:text-light dark:hover:bg-primary flex items-center rounded-md py-2 t
                                         <x-heroicons :name="$navigation['icon'] ?? 'no-symbol'" class="h-4 w-4" />
                                     </div>
                                 </div>
-                                <span class="truncate text-sm text-white"> {{ __($navigation['label']) }} </span>
-                                @if(count($navigation['children']) > 1)
+                                <span class="truncate text-sm text-white"> {{ __($navigation['label'] ?? $key) }} </span>
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply resources/views/livewire/navigation.blade.php:88]: headlining is done in the livewire class, i removed it

File: src/Actions/NotificationSetting/UpdateNotificationSetting.php [hunk 9]
```diff
@@ -14,7 +14,7 @@ class UpdateNotificationSetting extends BaseAction
     public function __construct(array $data)
     {
         parent::__construct(array_merge(['is_anonymous' => false], $data[0] ?? []));
-        $this->rules = $this->data['is_anonymous'] ?
+        $this->rules = ($this->data['is_anonymous'] ?? false) ?
```
Comment (SirSplasch) [inline src/Actions/NotificationSetting/UpdateNotificationSetting.php:17]: revert changes to avoid merge conflicts

**Review by SirSplasch — CHANGES_REQUESTED** [review]

**Pull request merged**